apps/api/src/routes/v1_keys_getWhoAmI.ts

```diff
@@ -0,0 +1,173 @@
+import type { App } from "@/pkg/hono/app";
+import { createRoute, z } from "@hono/zod-openapi";
+
+import { rootKeyAuth } from "@/pkg/auth/root_key";
+import { UnkeyApiError, openApiErrorResponses } from "@/pkg/errors";
+import { buildUnkeyQuery } from "@unkey/rbac";
+import { sha256 } from "@unkey/hash";
+
+const route = createRoute({
+  tags: ["keys"],
+  operationId: "whoAmI",
+  method: "get",
+  path: "/v1/keys.whoAmI",
+  security: [{ bearerAuth: [] }],
+  request: {
+    query: z.object({
+      key: z.string().min(1).openapi({
+        description: "The actual key to fetch",
+        example: "actual_key_value_here",
+      }),
+    }),
+  },
+  responses: {
+    200: {
+      description: "The configuration for a single key",
+      content: {
+        "application/json": {
+          schema:  z.object({
+            id: z.string().openapi({
+              description: "The ID of the key",
+              example: "abc123",
+            }),
+            name: z.string().optional().openapi({
+              description: "The name of the key",
+              example: "API Key 1",
+            }),
+            remaining: z.number().int().optional().openapi({
+              description: "The remaining number of requests for the key",
+              example: 1000,
+            }),
+            identity: z
+              .object({
+                id: z.string().openapi({
+                  description: "The identity ID associated with the key",
+                  example: "identity123",
```

Comment:
that's not what an identity id looks like, they're always prefixed with `id_`-> `id_Ly623bAsiHXwTfWPEVhEQ`